### DISCUSSION TITLE
Is there a way to append in the middle instead at the end when using useFieldArray append

### DISCUSSION THREAD

**Discussion by nimeshmaharjan1** [react-hook-form/react-hook-form#9673] [Q&A] [answered]
Right now I can append multiple form modals but when I duplicate a form modal I want it appended below it instead of at the end.

**Comment by Moshyfawn** [comment, marked as answer]
Try using [the `insert` field array method](https://react-hook-form.com/api/usefieldarray) instead

```js
const { insert } = useFieldArray()

insert(3, data) // INFO: (position index, field data)
```

**Reply by nimeshmaharjan1** [reply]
thank you so much!
